### PR TITLE
File operations return status code

### PR DESCRIPTION
**Pull request by hzeller**
Now that we use absl::Status in a lot of place, make the file operations return more detailed information.

### CODE REVIEW

File: common/util/file_util.cc [hunk 1]
```diff
@@ -50,20 +50,28 @@ absl::string_view Stem(absl::string_view filename) {
...
+      break;
+    case ENOENT:
+      translated_code = StatusCode::kNotFound;
+      break;
+    case EEXIST:
+      translated_code = StatusCode::kAlreadyExists;
+      break;
+    case EINVAL:
+      translated_code = StatusCode::kInvalidArgument;
+      break;
```
Comment (fangism) [inline common/util/file_util.cc:74]: worth declaring a function-local `static const map<int, StatusCode>*` for these?

Comment (hzeller) [reply common/util/file_util.cc:74]: I think it makes it more ugly and more slow, so wouldn't do it.

Comment (hzeller) [reply common/util/file_util.cc]: I've described it now more compactly. PTAL.

**Comment by hzeller** [comment]
(re-done in #295)

**Pull request closed without merging**